.circleci/config.yml

```diff
@@ -483,131 +309,101 @@ jobs:
-  #####################################
-
   build:
-    docker:
-      - image: cimg/go:1.17.2
-    resource_class: large
-    environment:
-      TMPDIR: /mnt/ramdisk
-    working_directory: /home/circleci/go/src/github.com/influxdata/influxdb
+    executor: cross-builder
+    parameters:
+      os:
+        type: string
+      arch:
+        type: string
+      build-type:
+        type: string
     steps:
       - checkout
-      - restore_cache:
-          name: Restore GOPATH/pkg/mod
-          keys:
-            - influxdb-gomod-sum-{{ checksum "go.sum" }}
-      - install_core_deps
-      - install_cross_bin_deps
-      # Build the static binary for linux
-      - run: goreleaser build --snapshot --single-target
-      - store_artifacts:
-          path: dist
-      - persist_to_workspace:
-          root: .
-          paths:
-            - project
-            - dist/influxd_linux_amd64/influxd
-            - etc/litmus_success_notify.sh
-            - etc/litmus_fail_notify.sh
       - run:
-          name: Exit the job with success if running on a fork
-          command: |
-            if [ -n "$CIRCLE_PR_NUMBER" ]; then
-                circleci-agent step halt
-            fi
-      - setup_remote_docker:
-          version: 19.03.13
-          docker_layer_caching: true
+          name: Generate UI assets
+          command: make generate-web-assets
       - run:
-          name: Build the candidate docker image
+          name: Build binaries
           command: |
-            cp dist/influxd_linux_amd64/influxd .
-            docker build -f docker/influxd/Dockerfile -t quay.io/influxdb/oss-acceptance:${CIRCLE_SHA1} .
-            docker save -o docker-image.tar quay.io/influxdb/oss-acceptance:${CIRCLE_SHA1}
+            export GOOS=<< parameters.os >>
+            export GOARCH=<< parameters.arch >>
+            ./scripts/ci/build.sh "bin/influxd_$(go env GOOS)_$(go env GOARCH)" "<< parameters.build-type >>" ./cmd/influxd
+      - store_artifacts:
+          path: bin
       - persist_to_workspace:
           root: .
           paths:
-            - docker-image.tar
-      - quay_login
-      - run:
-          name: Push the candidate image to quay
-          command: docker push quay.io/influxdb/oss-acceptance:${CIRCLE_SHA1}
+            - bin
 
-  cross_build:
-    executor: linux-amd64
-    environment:
-      TMPDIR: /mnt/ramdisk
+  package-and-release:
+    executor: cross-builder
+    parameters:
+      build-type:
+        type: string
     steps:
       - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          name: Compute artifact version
+          command: echo "export GORELEASER_CURRENT_TAG='$(./scripts/ci/build-version.sh << parameters.build-type >>)'" >> $BASH_ENV
+      - unless:
+          condition:
+            equal: [ release, << parameters.build-type >> ]
+          steps:
+            - run:
+                name: Tag commit locally to pass goreleaser validation
```

Comment:
I worry a little bit about running gorelease without the `--skip-validation` option since it ends up being kind of touchy about what things make the git checks fail. I do think its the right thing to do though, and hopefully as we port this to release branches (or adapt master to be able to do releases) we can keep the CI structured so that we are confident that if the nightly release passes the git checks, so too will a tagged release, so that there are no surprises when we try to do a tagged release.